genesyscloud/tfexporter/genesyscloud_resource_exporter.go

```diff
@@ -1552,6 +1551,11 @@ func (g *GenesysCloudResourceExporter) resourceIdExists(refID string, existingRe
 }
 
 func (g *GenesysCloudResourceExporter) isDataSource(resType string, name string) bool {
+	for _, element := range telephony_providers_edges_site.DataSourceManagedSites {
```

Comment:
this loop will increase linearly ("telephony_providers_edges_site.DataSourceManagedSites"),
 for every resource we might need to add as a datsource in the future. 
 
 suggest to create a datasource slice in exporter and refer it in telephony_providers_edges_site and make sure you lock it when appending data to it. this way we need not come to this place everytime we introduce a new resource  based on attribute as datasource.